### PR TITLE
Improvements and handle API errors

### PR DESCRIPTION
**Pull request by rgazelot**
Related to https://github.com/Calendart/CalendArt/pull/6

### CODE REVIEW

**Comment by rgazelot** [comment]
@Taluu Open for review

File: src/Api/CalendarApi.php [hunk 1]
```diff
@@ -28,6 +28,8 @@
  */
 class CalendarApi implements CalendarApiInterface
 {
+    use ResponseHandler;
```
Comment (Taluu) [inline src/Api/CalendarApi.php:30]: :+1:

**Comment by Taluu** [comment]
Keep in mind we must be compatible with php 5.4 (tbh, i think i'll scale up the dependency to 5.5...). And tests are failing because they're waiting on Calendar I guess, so I need to merge the other one first and relaunch tests

**Comment by rgazelot** [comment]
@Taluu Open for final review

**Comment by rgazelot** [comment]
@Taluu I fixed the php 5.4 stuffs

File: src/Api/ResponseHandler.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace CalendArt\Adapter\Office365\Api;
+
+use GuzzleHttp\Message\ResponseInterface;
+
+use CalendArt\Adapter\Office365\Exception;
+
+trait ResponseHandler
+{
+    protected function handleResponse(ResponseInterface $response)
```
Comment (Taluu) [inline src/Api/ResponseHandler.php]: could (should ?) be private

**Comment by Taluu** [comment]
Could you squash some commits ? Thanks

**Comment by Taluu** [comment]
:tada:

**Pull request merged**